src/fparser/common/splitline.py

```diff
@@ -249,94 +250,58 @@ def splitquote(line, stopchar=None, lower=False, quotechars="\"'"):
     current closing quotation character to be specified.
 
     :param str line: the line to split.
-    :param Optional[str] stopchar: the quote character that will terminate an \
-                                   existing quoted string or None otherwise.
-    :param bool lower: whether or not to convert the split parts of the line \
-                       to lowercase.
-    :param str quotechars: the characters that are considered to delimit \
-                           quoted strings.
-
-    :returns: tuple containing a list of the parts of the line split into \
-              those parts that are not quoted strings and those parts that are \
-              as well as the quote character corresponding with any quoted \
-              string that has not been closed before the end of the line.
-    :rtype: Tuple[List[str], str]
+    :param stopchar: the quote character that will terminate an
+                     existing quoted string or None otherwise.
+    :param lower: whether or not to convert the non-quoted parts of the line
+                  to lowercase.
+
+    :returns: tuple containing a list of the parts of the line split into
+              those parts that are not quoted strings and those parts that are
+              (as instances of String) as well as the quote character
+              corresponding with any quoted string that has not been closed
+              before the end of the line.
 
     """
-    # Will hold the various parts that `line` is split into.
-    items = []
-    # The current position in the line being processed.
-    ipos = 0
-    while 1:
-        # Move on to the next character in the line.
-        try:
-            char = line[ipos]
-            ipos += 1
-        except IndexError:
-            break
-        part = []
-        nofslashes = 0
-        if stopchar is None:
-            # search for string start
-            while 1:
-                if char in quotechars and not nofslashes % 2:
-                    # Found an un-escaped quote character.
-                    stopchar = char
-                    ipos -= 1
-                    # This marks the end of the current part.
-                    break
-                if char == "\\":
-                    nofslashes += 1
+    segments = []
+    i = 0
+    n = len(line)
+    quote_char = stopchar
+
+    while i < n:
```

Comment:
A comment or two would be useful:
`A quote character was found, collect all characters till closing quote, but ignore two consecutive quote characters`
OK, that's awful ;) But something like this.